ui-grid/ui-grid-tests.ts

```diff
@@ -1,20 +1,26 @@
 /// <reference path="ui-grid.d.ts" />
 /// <reference path="../angularjs/angular.d.ts" />
 
-var columnDef: uiGrid.IColumnDef;
+interface IMyEntity {
+    name: string;
+    age: number;
+}
+
+var columnDef: uiGrid.IColumnDefOf<IMyEntity>;
 columnDef.aggregationHideLabel = true;
 columnDef.aggregationHideLabel = false;
 columnDef.aggregationType = 1;
 columnDef.aggregationType = function () { return 1; };
 columnDef.cellClass = 'test';
-columnDef.cellClass = function (gridRow: uiGrid.IGridRow, gridCol: uiGrid.IGridColumn, index: number) {
-    return 'pizza';
+columnDef.cellClass = (gridRow, gridCol, index) => {
+    //types of gridRow, gridCol, and index are flowed in correctly
+    return `${gridRow.entity.name}-${gridCol.field}-${index + 1}`;
 };
 columnDef.cellFilter = 'date';
 columnDef.cellTemplate = '<div blah="something">hello</div>';
 columnDef.cellTooltip = 'blah';
 columnDef.cellTooltip = function (gridRow: uiGrid.IGridRow, gridCol: uiGrid.IGridColumn) {
-    return 'blah';
+    return `${gridRow.entity.unknownProperty}-${gridCol.displayName}`;
```

Comment:
@btesser, This is the potential harm when explicitly adding the type annotations on functions like these. You lose the generic type and could potentially introduce an error.  In this case the error is not as easily detected when searching the code than if you had made a cast to `any`.  IMHO it's best to let the function parameters get types automatically.